crates/dap/src/adapters.rs

```diff
@@ -244,49 +218,34 @@ pub trait DebugAdapter: 'static + Send + Sync {
         {
             log::info!("Using cached debug adapter binary {}", self.name());
 
-            return self.get_installed_binary(delegate, &config, None).await;
+            if let Some(binary) = self.get_installed_binary(delegate, &config).await.log_err() {
+                return Ok(binary);
+            }
+
```

Comment:
Nice edge case addition! We should also remove the curropted binary from `delegate.updated_adapters()` here.

That way if the fallback installation fails we don't have a cached binary that is likely to fail